### DISCUSSION TITLE
Timefold to 2d bin Packing

### DISCUSSION THREAD

**Discussion by leandroZanatta** [TimefoldAI/timefold-solver#460] [Ideas]
It is possible to create an example with timefold, for a cutting problem in sawmills, in a circular container, providing x,y,width,height positions of cuts, with cutting diameter restrictions

**Comment by ge0ffrey** [comment]
2D surface packing (or 3D volume packing) with Timefold Solver out of the box doesn't work well.

What you need to do is have a Local Search order a list of items, which then is the order in which they get assigned by a greedy Construction Heuristic and only then you can calculate the score.
Timefold Solver doesn't support that yet, nesting a CH in a LS score calculation.

That being said, you can try to hack it with an EasyScoreCalculator... I once did it for 3D volume packing for packing-santas-sleigh. You might find the code somewhere in my github repo's. Here be dragons. This is going off-track. Do not judge Timefold on this experience if you try this out.

**Comment by ge0ffrey** [comment]
2D bin packing with 2 unrelated dimensions (or more dimensions) works well out of the box with Timefold Solver.
